test/e2e/framework/framework.go

```
@@ -0,0 +1,274 @@
+    baseName string, labels map[string]string) *v1.Namespace {
+
+    ns:= createTestNamespace(clientSet, baseName, labels)
+    f.AddNamespacesToDelete(ns)
+    return ns
+}
+
+func (f *Framework) AddNamespacesToDelete(namespaces ...*v1.Namespace) {
+    for _, ns := range namespaces {
+        if ns == nil {
+            continue
+        }
+        f.namespacesToDelete = append(f.namespacesToDelete, ns)
+    }
+}
+
+func generateTestNamespace(client kubeclientset.Interface, baseName string, labels map[string]string) *v1.Namespace {
+    ginkgo.By("Generating a namespace to execute the tests in")
+    namespace := generateNamespace(client, baseName, labels)
+    ginkgo.By(fmt.Sprintf("Generated test namespace %s", namespace.Name))
+    return namespace
+}
+
+func generateNamespace(client kubeclientset.Interface, baseName string, labels map[string]string) *v1.Namespace {
+    namespaceObj := &corev1.Namespace{
+        ObjectMeta: metav1.ObjectMeta{
+            GenerateName: fmt.Sprintf("e2e-tests-%v-", baseName),
+            Labels: labels,
+        },
+    }
+
+    namespace, err := client.CoreV1().Namespaces().Create(namespaceObj)
+    Expect(err).NotTo(HaveOccurred(), "Error generating namespace %v", namespaceObj)
+    return namespace
+}
+
+func createTestNamespace(client kubeclientset.Interface, name string, labels map[string]string) *v1.Namespace {
+    ginkgo.By(fmt.Sprintf("Creating a namespace %s to execute the test in", name))
+    namespace := createNamespace(client, name, labels)
+    return namespace
+}
+
+func createNamespace(client kubeclientset.Interface, name string, labels map[string]string) *v1.Namespace {
+    namespaceObj := &corev1.Namespace{
+        ObjectMeta: metav1.ObjectMeta{
+            Name: name,
+            Labels: labels,
+        },
+    }
+
+    namespace, err := client.CoreV1().Namespaces().Create(namespaceObj)
+    Expect(err).NotTo(HaveOccurred(), "Error creating namespace %v", namespaceObj)
```

We should propagate the error as an invalid namespace will be returned.

---

same.